src/Tasks/CheckExternalLinksTask.php

```diff
@@ -0,0 +1,230 @@
+    protected $linkChecker;
+
+    protected $title = 'Checking broken External links in the SiteTree';
+
+    protected $description = 'A task that records external broken links in the SiteTree';
+
+    protected $enabled = true;
+
+    /**
+     * Log a message
+     *
+     * @param string $message
+     */
+    protected function log($message)
+    {
+        if (!$this->silent) {
+            Debug::message($message);
+        }
+    }
+
+    public function run($request)
+    {
+        $this->runLinksCheck();
+    }
+    /**
+     * Turn on or off message output
+     *
+     * @param bool $silent
+     */
+    public function setSilent($silent)
+    {
+        $this->silent = $silent;
+    }
+
+    /**
+     * @param LinkChecker $linkChecker
+     */
+    public function setLinkChecker(LinkChecker $linkChecker)
+    {
+        $this->linkChecker = $linkChecker;
+    }
+
+    /**
+     * @return LinkChecker
+     */
+    public function getLinkChecker()
+    {
+        return $this->linkChecker;
+    }
+
+    /**
+     * Check the status of a single link on a page
+     *
+     * @param BrokenExternalPageTrack $pageTrack
+     * @param DOMNode $link
+     */
+    protected function checkPageLink(BrokenExternalPageTrack $pageTrack, DOMNode $link)
+    {
+        $class = $link->getAttribute('class');
+        $href = $link->getAttribute('href');
+        $markedBroken = preg_match('/\b(ss-broken)\b/', $class);
+
+        // Check link
+        $httpCode = $this->linkChecker->checkLink($href);
+        if ($httpCode === null) {
+            return; // Null link means uncheckable, such as an internal link
+        }
+
+        // If this code is broken then mark as such
+        if ($foundBroken = $this->isCodeBroken($httpCode)) {
+            // Create broken record
+            $brokenLink = new BrokenExternalLink();
+            $brokenLink->Link = $href;
+            $brokenLink->HTTPCode = $httpCode;
+            $brokenLink->TrackID = $pageTrack->ID;
+            $brokenLink->StatusID = $pageTrack->StatusID; // Slight denormalisation here for performance reasons
+            $brokenLink->write();
+        }
+
+        // Check if we need to update CSS class, otherwise return
+        if ($markedBroken == $foundBroken) {
+            return;
+        }
+        if ($foundBroken) {
+            $class .= ' ss-broken';
+        } else {
+            $class = preg_replace('/\s*\b(ss-broken)\b\s*/', ' ', $class);
+        }
+        $link->setAttribute('class', trim($class));
+    }
+
+    /**
+     * Determine if the given HTTP code is "broken"
+     *
+     * @param int $httpCode
+     * @return bool True if this is a broken code
+     */
+    protected function isCodeBroken($httpCode)
+    {
+        // Null represents no request attempted
+        if ($httpCode === null) {
+            return false;
+        }
+
+        // do we have any whitelisted codes
+        $ignoreCodes = Config::inst()->get('CheckExternalLinks', 'IgnoreCodes');
```

Comment:
The CheckExternalLinks class probably needs to be namespaced for this code to work now